nn/transformer.py

```diff
@@ -0,0 +1,220 @@
+"""
+Transformer Modules
+"""
+
+from __future__ import annotations
+from .base import Module, LayerRef
+from .math_ import relu
+from ._generated_layers import Linear, layer_norm
```

Comment:
`_generated_layers`, `math_`, `utils`, `container`, etc are kind of implementation details which are not really necessary here. Just import from `.`.